underscore.js

```diff
@@ -521,8 +521,13 @@
   };
 
   // Flatten out an array, either recursively (by default), or just one level.
-  _.flatten = function(array, shallow) {
-    return flatten(array, shallow, false);
+  _.flatten = function(array, depth) {
+    if (!depth && depth !== 0) {
```

Comment:
the internal `flatten` doesn't handle depth 0 at all and will conflict with whatever "strict" is doing.
`depth` as with `shallow` is optional so the default value is `Infinity`, but to keep with the current API `false` should also be assumed as `Infinity` so I suppose it could be:
```js
if(depth === false || _.isUndefined(depth)) {
  depth = Infinity;
} else if (depth <= 0) {
  return _.clone(array);
}
```
We can certainly add the `else if (depth === true) {` but `flatten` is called internally with `true` elsewhere.